contracts/rewards/src/contract/execute.rs

```diff
@@ -0,0 +1,211 @@
+        _count: Option<u64>,
+    ) -> Result<HashMap<Addr, Uint256>, ContractError> {
+        todo!()
+    }
+
+    pub fn update_params(
+        &mut self,
+        _new_params: RewardsParams,
+        _block_height: u64,
+    ) -> Result<(), ContractError> {
+        todo!()
+    }
+
+    pub fn add_rewards(
+        &mut self,
+        _contract: Addr,
+        _amount: Uint256,
+        _block_height: u64,
+    ) -> Result<(), ContractError> {
+        todo!()
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use axelar_wasm_std::nonempty;
+    use cosmwasm_std::{Uint256, Uint64};
+
+    use crate::{
+        msg::RewardsParams,
+        state::{self, Epoch, StoredParams},
+    };
+
+    use super::Contract;
+
+    /// Tests that the current epoch is correctly computed when a number of blocks less than the epoch duration passes.
+    /// The epoch number should not increase.
+    #[test]
+    fn test_get_current_epoch_same_epoch() {
+        let (mut contract, current_epoch, stored_params) = setup();
+        let epoch_duration: u64 = stored_params.params.epoch_duration.into();
+        // epoch shouldn't change
+        let new_epoch = contract
+            .get_current_epoch(current_epoch.block_height_started + 1)
+            .unwrap();
+        assert_eq!(new_epoch, current_epoch);
+
+        // epoch still shouldn't change
+        let new_epoch = contract
+            .get_current_epoch(current_epoch.block_height_started + epoch_duration - 1)
+            .unwrap();
+        assert_eq!(new_epoch, current_epoch);
+    }
+
+    /// Tests that the current epoch is computed correctly when a number of blocks greater than or equal to the
+    /// epoch duration, but less than 2 times the epoch duration, passes. The epoch number should increase by one.
+    #[test]
+    fn test_get_current_epoch_next_epoch() {
+        let (mut contract, current_epoch, stored_params) = setup();
+        let epoch_duration: u64 = stored_params.params.epoch_duration.into();
+
+        // epoch should increase by one
+        let block = current_epoch.block_height_started + epoch_duration;
+        let new_epoch = contract.get_current_epoch(block).unwrap();
+        assert_ne!(new_epoch, current_epoch);
+        assert_eq!(
+            Epoch {
+                epoch_num: 2,
+                block_height_started: block,
+                rewards: current_epoch.rewards.clone()
+            },
+            new_epoch
+        );
+
+        // epoch should increase by one, but start of epoch is in the past
+        let block = current_epoch.block_height_started + epoch_duration + epoch_duration / 2;
+        let new_epoch = contract.get_current_epoch(block).unwrap();
+        assert_ne!(new_epoch, current_epoch);
+        assert_eq!(
+            Epoch {
+                epoch_num: 2,
+                block_height_started: block - epoch_duration / 2,
+                rewards: current_epoch.rewards.clone()
+            },
+            new_epoch
+        );
+    }
```

Comment:
legibility of this test can be improved by parameterization, something like this:
```
let test_cases = vex![(height1, expected_num1, expected_start1), (height2, expected_num2, expected_start2)]
for case in test_cases.iter(){
   // assertions
}
```